pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,295 @@
+
+	fmt.Println("Waiting for operator pod to stop")
+	err = debug.SetDeploymentScale(context, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		return fmt.Errorf("failed to stop deployment rook-ceph-operator. %v", err)
+	}
+	fmt.Println("rook-ceph-operator deployment scaled down")
+
+	fmt.Println("Waiting for bad mon pod to stop")
+	for _, badMon := range badMons {
+		err = debug.SetDeploymentScale(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", badMon), 0)
+		if err != nil {
+			return fmt.Errorf("deployment %s still exist. %v", fmt.Sprintf("rook-ceph-mon-%s", badMon), err)
+		}
+		fmt.Printf("deployment.apps/%s scaled\n", fmt.Sprintf("rook-ceph-mon-%s", badMon))
+	}
+
+	debug.StartDebug(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", goodMon), "")
+
+	debugDeploymentSpec, err := debug.GetDeployment(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-mon-%s-debug", goodMon)
+	}
+
+	labelSelector := fmt.Sprintf("ceph_daemon_type=%s,ceph_daemon_id=%s", debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_type"], debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_id"])
+	_, err = k8sutil.WaitForPodToRun(context, clusterNamespace, labelSelector)
+	if err != nil {
+		return fmt.Errorf("failed to start deployment %s", fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	}
+
+	UpdateMonMap(context, clusterNamespace, labelSelector, cephFsid, goodMon, goodMonPublicIp, badMons)
+
+	fmt.Println("Restoring the mons in the rook-ceph-mon-endpoints configmap to the good mon")
+	monCm.Data["data"] = fmt.Sprintf("%s=%s:%s", goodMon, goodMonPublicIp, goodMonPort)
+
+	monCm, err = context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Update(ctx.TODO(), monCm, v1.UpdateOptions{})
+	if err != nil {
+		log.Fatalf("failed to update mon configmap %s %v", MonConfigMap, err)
+	}
+
+	fmt.Printf("Stopping the debug pod for mon %s.\n", goodMon)
+	debug.StopDebug(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", goodMon))
+
+	fmt.Println("Check that the restored mon is responding")
+	err = waitForMonStatusResponse(context, clusterNamespace)
+	if err != nil {
+		return err
+	}
+
+	err = removeBadMonsResources(context, clusterNamespace, badMons)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Mon quorum was successfully restored to mon %s\n", goodMon)
+	fmt.Println("Only a single mon is currently running")
+
+	output, err = promptToContinueOrCancel(answer)
+	if err != nil {
+		return fmt.Errorf(" restoring the mon quorum to mon %s cancelled", goodMon)
+	}
+	fmt.Println(output)
+
+	err = debug.SetDeploymentScale(context, clusterNamespace, "rook-ceph-operator", 1)
+	if err != nil {
+		return fmt.Errorf("failed to start deployment rook-ceph-operator. %v", err)
+	}
+
+	return nil
+}
+
+func UpdateMonMap(context *k8sutil.Context, clusterNamespace, labelSelector, cephFsid, goodMon, goodMonPublicIp string, badMons []string) {
```

Comment:
use lower case methods whenever the method is not needed outside the package
```suggestion
func updateMonMap(context *k8sutil.Context, clusterNamespace, labelSelector, cephFsid, goodMon, goodMonPublicIp string, badMons []string) {
```

Reply:
If this function fails, it will just exit, right?

Reply:
yes, since we are passing `true` in exitOnError